linenoise.c

```diff
@@ -794,10 +824,54 @@ static int linenoiseEdit(int stdin_fd, int stdout_fd, char *buf, size_t buflen,
     if (write(l.ofd,prompt,l.plen) == -1) return -1;
     while(1) {
         char c;
-        int nread;
+        int nread = 0;
         char seq[3];
 
-        nread = read(l.ifd,&c,1);
+        fd_set rfds;
+        int retval;
+
+        FD_ZERO(&rfds);
+        FD_SET(l.ifd, &rfds);
+        FD_SET(ctrl_pipe_read_fd, &rfds);
+        int nfds = (l.ifd > ctrl_pipe_read_fd) ? l.ifd+1 : ctrl_pipe_read_fd+1; //compute 1+(the greater fd) for select
+
+        struct timeval timeout;
+        timeout.tv_sec = 0;
+        timeout.tv_usec = 1e5; // 0.1 seconds
```

Comment:
`1e5` is a floating point literal (a double) which I felt like commenting on but will be converted implicitly by the compiler and I'm really far into this sentence and still can't think of any reason this could be problematic.